src/shared/telemetry/telemetryUtils.ts

```diff
@@ -21,11 +34,13 @@ export function registerCommand<T>({
     command,
     thisArg,
     register = vscode.commands.registerCommand,
-    callback
+    telemetryName,
+    callback,
 }: {
     command: string
     thisArg?: any
     register?: typeof vscode.commands.registerCommand
+    telemetryName?: TelemetryName
```

Comment:
You can simplify this with:

```typescript
export function registerCommand<T>({
    command,
    ...
    telemetryName = {
        namespace: 'Command',
        name: command
    },
    ...
}: {
    ...
}) {
    ...
    const datum = result && result.datum ? result.datum : defaultMetricDatum(telemetryName.name)
    ...
    ext.telemetry.record({
        namespace: telemetryName.namespace,
        ...
    })
    ...
}
```

Reply:
So this won't work without a little futzing: `Command` is not a valid enum in `TelemetryNamespace`, but is instead there to preserve existing functionality (this goes for the `command` as `name` as well).

I don't want to make this an enum (since this isn't officially-recognized by the system) but I don't want to toss the existing metrics before we consciously do something with them.

Thoughts?

Reply:
✔️